From: Michael Davidsaver <mdavidsaver@bnl.gov>
Date: Wed, 19 Sep 2012 10:23:46 -0400
Subject: allow more flexible makefile config

---
 configure/CONFIG      | 7 +++++++
 configure/RULES_BUILD | 5 +++++
 2 files changed, 12 insertions(+)

diff --git a/configure/CONFIG b/configure/CONFIG
index 8b7b785..2d1f486 100644
--- a/configure/CONFIG
+++ b/configure/CONFIG
@@ -114,6 +114,13 @@ ifneq ($(TOP_CFG_CONFIGS),)
   include $(TOP_CFG_CONFIGS)
 endif
 
+#---------------------------------------------------------------
+# Allow installed 3rd party modules to add/modify make variables
+
+-include $(wildcard $(CONFIG)/conf.d/*.make)
+# old location
+-include $(wildcard $(EPICS_BASE)/configure/conf.d/*.make)
+
 #  User specific definitions
 #
 -include $(HOME)/configure/CONFIG_USER
diff --git a/configure/RULES_BUILD b/configure/RULES_BUILD
index 1c38eef..8833086 100644
--- a/configure/RULES_BUILD
+++ b/configure/RULES_BUILD
@@ -515,6 +515,11 @@ $(INSTALL_TEMPLATES_SUBDIR)/%: %
 
 -include $(CONFIG)/RULES_EXPAND
 
+# Allow 3rd party modules to inject additional rules
+-include $(wildcard $(CONFIG)/rules.d/*.make)
+# old location
+-include $(wildcard $(EPICS_BASE)/configure/rules.d/*.make)
+
 .PRECIOUS: %.i %.o %.c %.nm %.cpp %.cc
 .PRECIOUS: $(COMMON_INC)
 
