From: Michael Davidsaver <mdavidsaver@bnl.gov>
Date: Wed, 19 Sep 2012 10:35:00 -0400
Subject: install makeInstallDir.pl

---
 src/std/softIoc/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/std/softIoc/Makefile b/src/std/softIoc/Makefile
index a432d35..948a2f1 100644
--- a/src/std/softIoc/Makefile
+++ b/src/std/softIoc/Makefile
@@ -12,6 +12,8 @@ SRC_DIRS += $(STDDIR)/softIoc
 PROD_IOC_DEFAULT = softIoc
 PROD_IOC_iOS = -nil-
 
+PERL_SCRIPTS += makeInstallDir.pl
+
 DBD += base.dbd
 DBD += asSub.dbd
 DBD += softIoc.dbd
