From: Michael Davidsaver <mdavidsaver@bnl.gov>
Date: Wed, 19 Sep 2012 10:46:02 -0400
Subject: for init scripts, ensure abs. path is used.

---
 src/ca/client/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ca/client/Makefile b/src/ca/client/Makefile
index fcbc574..a295cae 100644
--- a/src/ca/client/Makefile
+++ b/src/ca/client/Makefile
@@ -114,4 +114,4 @@ OBJS_vxWorks += ca_test
 include $(TOP)/configure/RULES
 
 S99%: ../rc2.%
-	sed -e s%:INSTALL_BIN:%`cd $(INSTALL_BIN); pwd`% $< >$@
+	sed -e s%:INSTALL_BIN:%$(FINAL_LOCATION)/bin/$(T_A)% $< >$@
