From: Michael Davidsaver <mdavidsaver@gmail.com>
Date: Thu, 10 Mar 2016 13:10:17 -0500
Subject: only -D an -I in pkg-config files

---
 src/tools/Makefile | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/tools/Makefile b/src/tools/Makefile
index e15e935..6d22065 100644
--- a/src/tools/Makefile
+++ b/src/tools/Makefile
@@ -60,8 +60,9 @@ HTMLS = style.css
 # Build Package Config Files
 
 FINAL_LOCATION ?= $(shell $(PERL) $(TOOLS)/fullPathName.pl $(INSTALL_LOCATION))
-C_CFLAGS += $(filter-out -g,$(filter-out -O%,$(filter-out -W%,$(CPPFLAGS))))
-C_CFLAGS += $(filter-out -g,$(filter-out -O%,$(filter-out -W%,$(CFLAGS))))
+X_CFLAGS = $(filter-out $(HARDEN_CPPFLAGS) $(HARDEN_CFLAGS) $(HARDEN_CXXFLAGS),$(CFLAGS) $(CPPFLAGS))
+C_CFLAGS += $(filter -I%,$(X_CFLAGS))
+C_CFLAGS += $(filter -D%,$(X_CFLAGS))
 PKGVARS += FINAL_LOCATION OS_CLASS CMPLR_CLASS C_CFLAGS LDFLAGS LDLIBS
 PKGVARS += EPICS_VERSION EPICS_REVISION EPICS_MODIFICATION EPICS_PATCH_LEVEL
 PKGVARS += CC CCC CPP AR LD
