From: Michael Davidsaver <mdavidsaver@bnl.gov>
Date: Wed, 19 Sep 2012 10:35:00 -0400
Subject: install makeInstallDir.pl

---
 src/softIoc/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/softIoc/Makefile b/src/softIoc/Makefile
index c3e8679..8b8ec34 100644
--- a/src/softIoc/Makefile
+++ b/src/softIoc/Makefile
@@ -12,6 +12,8 @@ include $(TOP)/configure/CONFIG
 PROD_IOC_DEFAULT = softIoc
 PROD_IOC_iOS = -nil-
 
+PERL_SCRIPTS += makeInstallDir.pl
+
 DBD += softIoc.dbd
 softIoc_DBD += base.dbd
 softIoc_DBD += dlload.dbd
