From: Michael Davidsaver <mdavidsaver@bnl.gov>
Date: Wed, 19 Sep 2012 10:46:02 -0400
Subject: [PATCH] for init scripts, ensure abs. path is used.

---
 src/util/Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/util/Makefile b/src/util/Makefile
index e0cd22c..37ef4ed 100644
--- a/src/util/Makefile
+++ b/src/util/Makefile
@@ -33,6 +33,6 @@ SCRIPTS_Linux := S99logServer S99caRepeater
 include $(TOP)/configure/RULES
  
 S99%: ../rc2.%
-	sed -e s%:INSTALL_BIN:%`cd $(INSTALL_BIN); pwd`% $< >$@
+	sed -e s%:INSTALL_BIN:%$(FINAL_LOCATION)/bin/$(T_A)% $< >$@
 
 #	EOF Makefile.Host for base/src/util
-- 
