From: Martin Konrad <konrad@frib.msu.edu>
Date: Tue, 13 Oct 2020 18:38:17 -0400
Subject: Simplify addToList() function

---
 src/ioc/db/dbScan.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/ioc/db/dbScan.c b/src/ioc/db/dbScan.c
index fc3d1dd..984ba8a 100644
--- a/src/ioc/db/dbScan.c
+++ b/src/ioc/db/dbScan.c
@@ -1052,13 +1052,10 @@ static void addToList(struct dbCommon *precord, scan_list *psl)
     pse->pscan_list = psl;
     ptemp = (scan_element *)ellLast(&psl->list);
     while (ptemp) {
-        if (ptemp->precord->phas <= precord->phas) {
-            ellInsert(&psl->list, &ptemp->node, &pse->node);
-            break;
-        }
+        if (ptemp->precord->phas <= precord->phas) break;
         ptemp = (scan_element *)ellPrevious(&ptemp->node);
     }
-    if (ptemp == NULL) ellInsert(&psl->list, NULL, &pse->node);
+    ellInsert(&psl->list, (ptemp ? &ptemp->node : NULL), &pse->node);
     psl->modified = TRUE;
     epicsMutexUnlock(psl->lock);
 }
