From: Dirk Zimoch <dirk.zimoch@psi.ch>
Date: Wed, 28 Aug 2019 15:23:00 +0200
Subject: send pending log messages directly after connecting

---
 src/libCom/log/logClient.c | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/src/libCom/log/logClient.c b/src/libCom/log/logClient.c
index 203f29d..743910f 100644
--- a/src/libCom/log/logClient.c
+++ b/src/libCom/log/logClient.c
@@ -223,7 +223,7 @@ void epicsShareAPI logClientFlush ( logClientId id )
 
     logClient * pClient = ( logClient * ) id;
 
-    if ( ! pClient ) {
+    if ( ! pClient || ! pClient->connected ) {
         return;
     }
 
@@ -419,12 +419,8 @@ static void logClientRestart ( logClientId id )
 
         epicsMutexUnlock ( pClient->mutex );
 
-        if ( isConn ) {
-            logClientFlush ( pClient );
-        }
-        else {
-            logClientConnect ( pClient );
-        }
+        if ( ! isConn ) logClientConnect ( pClient );
+        logClientFlush ( pClient );
 
         epicsEventWaitWithTimeout ( pClient->shutdownNotify, LOG_RESTART_DELAY);
 
