From: Ralph Lange <Ralph.Lange@gmx.de>
Date: Thu, 26 Sep 2013 15:23:46 +0200
Subject: Fix [-Werror=format-security] issues in cap5

---
 src/cap5/Cap5.xs | 30 +++++++++++++++---------------
 1 file changed, 15 insertions(+), 15 deletions(-)

diff --git a/src/cap5/Cap5.xs b/src/cap5/Cap5.xs
index c3e4d0e..71ca527 100644
--- a/src/cap5/Cap5.xs
+++ b/src/cap5/Cap5.xs
@@ -461,7 +461,7 @@ SV * CA_new(const char *class, const char *name, ...) {
         SvREFCNT_dec(ca_ref);
         if (pch->conn_sub)
             SvREFCNT_dec(pch->conn_sub);
-        croak(get_error_msg(status));
+        croak("%s", get_error_msg(status));
     }
 
     return ca_ref;
@@ -490,7 +490,7 @@ void CA_DESTROY(SV *ca_ref) {
     Safefree(pch);
 
     if (status != ECA_NORMAL)
-        croak(get_error_msg(status));
+        croak("%s", get_error_msg(status));
 }
 
 
@@ -515,7 +515,7 @@ void CA_change_connection_event(SV *ca_ref, SV *sub) {
     status = ca_change_connection_event(pch->chan, handler);
 
     if (status != ECA_NORMAL) {
-        croak(get_error_msg(status));
+        croak("%s", get_error_msg(status));
     }
 }
 
@@ -599,7 +599,7 @@ void CA_put(SV *ca_ref, SV *val, ...) {
         Safefree(p.dbr);
     }
     if (status != ECA_NORMAL) {
-        croak(get_error_msg(status));
+        croak("%s", get_error_msg(status));
     }
     XSRETURN(0);
 }
@@ -694,7 +694,7 @@ void CA_put_callback(SV *ca_ref, SV *sub, SV *val, ...) {
     }
     if (status != ECA_NORMAL) {
         SvREFCNT_dec(put_sub);
-        croak(get_error_msg(status));
+        croak("%s", get_error_msg(status));
     }
     XSRETURN(0);
 }
@@ -735,7 +735,7 @@ void CA_put_acks(SV *ca_ref, SV *sevr, ...) {
         status = ca_put(DBR_PUT_ACKS, pch->chan, &acks);
 
     if (status != ECA_NORMAL)
-        croak(get_error_msg(status));
+        croak("%s", get_error_msg(status));
 
     XSRETURN(0);
 }
@@ -759,7 +759,7 @@ void CA_put_ackt(SV *ca_ref, int ack, ...) {
         status = ca_put(DBR_PUT_ACKS, pch->chan, &ackt);
 
     if (status != ECA_NORMAL)
-        croak(get_error_msg(status));
+        croak("%s", get_error_msg(status));
 
     XSRETURN(0);
 }
@@ -787,7 +787,7 @@ void CA_get(SV *ca_ref) {
         status = ca_get(best_type(pch), pch->chan, &pch->data);
     }
     if (status != ECA_NORMAL) {
-        croak(get_error_msg(status));
+        croak("%s", get_error_msg(status));
     }
 }
 
@@ -868,7 +868,7 @@ void CA_get_callback(SV *ca_ref, SV *sub, ...) {
 
 exit_croak:
     SvREFCNT_dec(get_sub);
-    croak(croak_msg);
+    croak("%s", croak_msg);
 }
 
 
@@ -950,7 +950,7 @@ SV * CA_create_subscription(SV *ca_ref, const char *mask_str, SV *sub, ...) {
 exit_croak:
     SvREFCNT_dec(mon_ref);
     SvREFCNT_dec(mon_sub);
-    croak(croak_msg);
+    croak("%s", croak_msg);
 }
 
 
@@ -967,7 +967,7 @@ void CA_clear_subscription(const char *class, SV *mon_ref) {
     status = ca_clear_subscription(event);
 
     if (status != ECA_NORMAL) {
-        croak(get_error_msg(status));
+        croak("%s", get_error_msg(status));
     }
 }
 
@@ -977,7 +977,7 @@ void CA_clear_subscription(const char *class, SV *mon_ref) {
 void CA_pend_io(const char *class, double timeout) {
     int status = ca_pend_io(timeout);
     if (status != ECA_NORMAL) {
-        croak(get_error_msg(status));
+        croak("%s", get_error_msg(status));
     }
 }
 
@@ -992,7 +992,7 @@ int CA_test_io(const char *class) {
 void CA_pend_event(const char *class, double timeout) {
     int status = ca_pend_event(timeout);
     if (status != ECA_TIMEOUT) {
-        croak(get_error_msg(status));
+        croak("%s", get_error_msg(status));
     }
 }
 
@@ -1080,7 +1080,7 @@ void CA_add_exception_event(const char *class, SV *sub) {
     if (status != ECA_NORMAL) {
         SvREFCNT_dec(exception_sub);
         exception_sub = NULL;
-        croak(get_error_msg(status));
+        croak("%s", get_error_msg(status));
     }
 }
 
@@ -1138,7 +1138,7 @@ void CA_replace_printf_handler(const char *class, SV *sub) {
     if (status != ECA_NORMAL) {
         SvREFCNT_dec(printf_sub);
         printf_sub = NULL;
-        croak(get_error_msg(status));
+        croak("%s", get_error_msg(status));
     }
 }
 
