From: Dirk Zimoch <dirk.zimoch@psi.ch>
Date: Tue, 17 Sep 2019 11:59:02 +0200
Subject: removed unneeded include

---
 src/libCom/log/logClient.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/libCom/log/logClient.c b/src/libCom/log/logClient.c
index 72a2e13..723f5d4 100644
--- a/src/libCom/log/logClient.c
+++ b/src/libCom/log/logClient.c
@@ -25,7 +25,6 @@
 #include "dbDefs.h"
 #include "epicsEvent.h"
 #include "iocLog.h"
-#include "errlog.h"
 #include "epicsMutex.h"
 #include "epicsThread.h"
 #include "epicsTime.h"
