From: Jeff Hill <johill@lanl.gov>
Date: Thu, 17 Oct 2013 09:23:47 -0600
Subject: fixed potential memory leak

---
 src/cas/generic/casStrmClient.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/cas/generic/casStrmClient.cc b/src/cas/generic/casStrmClient.cc
index 51f4d0d..5bcefb6 100644
--- a/src/cas/generic/casStrmClient.cc
+++ b/src/cas/generic/casStrmClient.cc
@@ -884,7 +884,7 @@ caStatus casStrmClient::monitorResponse (
             }
             else {
                 ecaStatus = ECA_GETFAIL;
-            }                
+            }               
             return monitorFailureResponse ( guard, msg, ecaStatus );
         }
         else {
@@ -926,6 +926,7 @@ caStatus casStrmClient::monitorResponse (
     int cacStatus = caNetConvert ( 
         msg.m_dataType, pPayload, pPayload, true, msg.m_count );
     if ( cacStatus != ECA_NORMAL ) {
+        pDBRDD->unreference ();
         return this->sendErrWithEpicsStatus ( 
             guard, & msg, chan.getCID(), S_cas_internal, cacStatus );
     }
