From: Dirk Zimoch <dirk.zimoch@psi.ch>
Date: Thu, 19 Sep 2019 11:42:04 +0200
Subject: bugfix: memmove'ed to much

---
 src/libCom/log/logClient.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libCom/log/logClient.c b/src/libCom/log/logClient.c
index 28b933e..4e647bd 100644
--- a/src/libCom/log/logClient.c
+++ b/src/libCom/log/logClient.c
@@ -294,10 +294,10 @@ void epicsShareAPI logClientFlush ( logClientId id )
     else if ( nSent > 0 && pClient->nextMsgIndex > 0 ) {
         pClient->backlog = epicsSocketCountUnsentBytes ( pClient->sock );
         nSent -= pClient->backlog;
-        if ( nSent > 0 ) {
+        pClient->nextMsgIndex -= nSent;
+        if ( nSent > 0 && pClient->nextMsgIndex > 0 ) {
             memmove ( pClient->msgBuf, & pClient->msgBuf[nSent],
                 pClient->nextMsgIndex );
-            pClient->nextMsgIndex -= nSent;
         }
     }
     epicsMutexUnlock ( pClient->mutex );
