From: Florian Feldbauer <florian@ep1.rub.de>
Date: Mon, 3 Nov 2014 11:54:26 -0500
Subject: Remove definition of cross compiler for linux-arm

---
 configure/os/CONFIG.Common.linux-arm | 24 ------------------------
 1 file changed, 24 deletions(-)

diff --git a/configure/os/CONFIG.Common.linux-arm b/configure/os/CONFIG.Common.linux-arm
index 15dad9e..7cfa8f5 100644
--- a/configure/os/CONFIG.Common.linux-arm
+++ b/configure/os/CONFIG.Common.linux-arm
@@ -10,27 +10,3 @@
 include $(CONFIG)/os/CONFIG.Common.linuxCommon
 
 ARCH_CLASS = arm
-
-ifeq ($(BUILD_CLASS),CROSS)
-    VALID_BUILDS = Ioc
-    GNU_TARGET = arm-linux
-
-    # prefix of compiler tools
-    CMPLR_SUFFIX =
-    CMPLR_PREFIX = $(addsuffix -,$(GNU_TARGET))
-
-    # Provide a link-time path for shared libraries
-    SHRLIBDIR_RPATH_LDFLAGS_YES += $(SHRLIB_DEPLIB_DIRS:%=-Wl,-rpath-link,%)
-    SHRLIBDIR_LDFLAGS += $(SHRLIBDIR_RPATH_LDFLAGS_$(LINKER_USE_RPATH))
-
-    # Provide a link-time path for products
-    PRODDIR_RPATH_LDFLAGS_YES += $(PROD_DEPLIB_DIRS:%=-Wl,-rpath-link,%)
-    PRODDIR_LDFLAGS += $(PRODDIR_RPATH_LDFLAGS_$(LINKER_USE_RPATH))
-
-    # Provide a link-time path for readline 
-    RUNTIME_LDFLAGS_READLINE_YES = -Wl,-rpath-link,$(GNU_DIR)/lib
-    RUNTIME_LDFLAGS_READLINE = $(RUNTIME_LDFLAGS_READLINE_$(LINKER_USE_RPATH))
-    RUNTIME_LDFLAGS_READLINE_CURSES = $(RUNTIME_LDFLAGS_READLINE_$(LINKER_USE_RPATH))
-    RUNTIME_LDFLAGS_READLINE_NCURSES = $(RUNTIME_LDFLAGS_READLINE_$(LINKER_USE_RPATH))
-
-endif
