From: Michael Davidsaver <mdavidsaver@bnl.gov>
Date: Tue, 17 Feb 2015 10:57:05 -0500
Subject: dbReadTest.c: flush errlog on exit as per lp:1419879

---
 src/dbStatic/dbReadTest.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/dbStatic/dbReadTest.c b/src/dbStatic/dbReadTest.c
index c7a4332..d377101 100644
--- a/src/dbStatic/dbReadTest.c
+++ b/src/dbStatic/dbReadTest.c
@@ -86,5 +86,6 @@ int main(int argc,char **argv)
     dbReportDeviceConfig(pdbbase,stdout);
 */
     dbFreeBase(pdbbase);
+    errlogFlush();
     return(0);
 }
